### PR TITLE
Item Plants, Grown Weapons, Botany Fixes/Changes

### PR DESCRIPTION
**Pull request by FalseIncarnate**
Botany plants can grow preset items, mobs, or fruit
- Turns out that the code for mob harvestables could be used almost unchanged to support item harvestables
 - Changed some variable names to better reflect this

Re-introduces the Egg Plant!
- Grow eggs to throw at the security officer who confiscated your ambrosia!
 - Exactly the same as eggs laid by chickens, except they shouldn't hatch
- Mutant species of Eggplants

Adjusts code and logic order to avoid a conflict with TRAIT_SPREAD fruits that can be converted into other items
- Grass, aloe, and comfrey were either unable to be converted or unable to be planted outside a tray if they had TRAIT_SPREAD due to logic ordering
- You must now be in DISARM intent to plant spreading fruits
 - Being in HELP or GRAB will attempt to convert the fruit into a poultice or other item as appropriate.
 - If the plant does not have TRAIT_SPREAD, you may also use DISARM to accomplish this

Adds Money Trees!
- Literally grows money-filled fruit called Cashpods
- Fruit contains gold and silver reagents, or can be turned into space cash by using it in-hand like grass or aloe
 - Value of the resulting space cash is related to potency. Higher potency = higher dosh!
 - Eating the fruit doesn't give you money, but may fill you with guilt (not really).
- Mutant species of lemons.
 - When life gives you lemons...

Adds the ability to convert sunflowers and novaflowers into their "weapon" forms
- Done exactly the same as converting grass, aloe, comfrey, or cashpods (use in-hand on help/grab intent)
 - Sunflower conversion is silent, though the item description changes
 - Novaflower conversion is not silent. You'll know when it happens.
- Hitting someone with a sunflower (weapon form) or novaflower (weapon form) will assault both parties with a green and yellow message.
 - FLOWER POWER!
- Hitting someone with a novaflower (weapon form) will increase their body temperature, and may cause burns as a result
- Picking up a novaflower (weapon form) without gloves will cause burns, but doesn't drop the flower.

Poppies now use modular product icons again
- This means you can mutate and splice your poppies to grow different colored flowers
 - This is one of the few modular icons that was interestingly almost exactly the same as the preset

COMPLETELY DELETES the code\objects\items\weapons\hydroponics.dm file
- This file was 90% commented out already, and I moved the remaining code into code\modules\hydroponics\grown_inedible.dm for organization
 - This affects the .dme by removing a single line as the file is no longer included.

### CODE REVIEW

File: code/modules/hydroponics/grown_inedible.dm [hunk 1]
```diff
@@ -84,3 +84,47 @@
...
+	throwforce = 0
+	throw_speed = 4
+	throw_range = 20
+
+/obj/item/weapon/grown/novaflower/attack(mob/living/carbon/M as mob, mob/user as mob)
+	M << "<font color='green'>[user] smacks you with a [name]!</font><font color='yellow'><b>FLOWER POWER</b></font>"
+	user << "<font color='green'> Your [name]'s </font><font color='yellow'><b>FLOWER POWER</b></font><font color='green'> strikes [M]</font>"
+	if(istype(M, /mob/living))
+		M << "<span class='warning'>You are heated by the warmth of the of the [name]!</span>"
+		M.bodytemperature += potency/2 * TEMPERATURE_DAMAGE_COEFFICIENT
```
Comment (Fox-McCloud) [inline code/modules/hydroponics/grown_inedible.dm]: TG changed this to actually adjust fire stacks and ignite the mob---with more stacks being applied based on potency--likewise, it loses endurance each time you whack someone with it, with the plant deleting itself when the endurance drops to 0.

**Comment by Fox-McCloud** [comment]
:+1: 

Though, not adding SLOT_HEAD for the flowers (and their on mob icons) = shameful.

**Comment by FalseIncarnate** [comment]
I didn't make the nova flower use fire stacks like TG because it was not a port but rather an update of existing code. Further, we already have lots of ways to light someone on fire, but few to affect body temperature directly. This makes nova flowers have non-harmful uses (counter cold environments), which although rare, couldn't be done with fire stacks.

As for the flowers and head slots, that was more due to lacking on mob sprites for head slot flowers (also they'd ideally be ear slot). Would be strange to see a flower listed as being equipped to the head/ear, yet actually appear in hand.

The "praise the sun" message only appears once per nova flower, and the process cannot be reversed, meaning you'd need to be able to rapid-fire pickup, activate, drop, and repeat in large quantities to make it any spam more than a copy/pasted message.

**Comment by FalseIncarnate** [comment]
Oh, tiny change I forgot to document (will add to original post later) is that I made poppies use modular product sprites again.

This will let enterprising botanists create a veritible rainbow of flowers to then bake into muffins.

**Comment by Fox-McCloud** [comment]
>The "praise the sun" message only appears once per nova flower, and the process cannot be reversed, meaning you'd need to be able to rapid-fire pickup, activate, drop, and repeat in large quantities to make it any spam more than a copy/pasted message.

I'm aware; that's exactly what I'm concerned about though--considering these things are pretty useless as a non-weapon, it's likely that people are going to do large conversion of them in a short order---that's going to get stupidly spammy to deadchat and anyone around that person.

**Comment by FalseIncarnate** [comment]
While I don't think that's going to be the case, especially given that this message only appears on novaflowers, which are a mutant species and have a rather low starting yield, I am willing to make a compromise and have the actual shouting of the message be a small chance, and have times you don't shout it instead display it as a message to the user themselves to maintain the purpose of the message: to inform you when you've converted the fruit to a weapon.

**Comment by Fox-McCloud** [comment]
One last request.

Mind making it so that when you splat something onto a turf, it applies the content to the turf?

**Comment by FalseIncarnate** [comment]
I'll do that in a separate PR

**Pull request merged**